release: Final changes for the 3.5.0 release

jhedberg opened the pull request:
Update the version index in the documentation, and mark the final release in the `VERSION` file.

Commit:
doc: add 3.5.0 to the index of releases

Add the new release to the doc index.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
release: Zephyr 3.5.0 release

Update the version for the final release.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

kartben reviewed (changes requested):
I think this PR should also have the commit that updates the list of supported releases in https://docs.zephyrproject.org/latest/releases/index.html#supported-releases (and that it should be added to the checklist to update it as part of the release) so that the information is captured in the tagged version.

Commit:
doc: release: add Zephyr 3.5 as supported release

Add Zephyr 3.5 as a supported release until July 26th 2024, i.e. when
3.7 is planned to be released.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>
